include/asterisk/res_pjsip.h

```diff
@@ -3500,6 +3514,53 @@ struct ast_threadpool *ast_sip_threadpool(void);
  */
 struct ast_sip_transport_state *ast_sip_get_transport_state(const char *transport_id);
 
+/*!
+ * \brief Return the SIP URI of the Contact header
+ * 
+ * @param tdata
+ * \retval Pointer to SIP URI of Contact
+ * \retval NULL if Contact header not found or not a SIP(S) URI
+ *
+ * \note Do not free the returned object.
+ */
+pjsip_sip_uri *ast_sip_get_contact_sip_uri(pjsip_tx_data *tdata);
+
+/*!
+ * \brief Returns the transport state currently in use based on request transport details
+ *
+ * @param details
```

Comment:
Same here regarding doxygen, and other places

Reply:
Done.